MoveSorter.hpp

```diff
@@ -54,11 +54,8 @@ class MoveSorter {
    * @return next remaining move with max score and remove it from the container.
    * If no more move is available return 0
    */
-  Position::position_t getNext() {
-    if(size)
-      return entries[--size].move;
-    else
-      return 0;
+  inline Position::position_t getNext() __attribute__((always_inline)) {
+    return size ? entries[--size].move : 0;
   }
```

Comment:
I prefer avoiding non standard C++ code.

moreover any good C++ compiler will almost certainly inline this code without having to force it.

Reply:
http://www.cplusplus.com/articles/1AUq5Di1/
^This is standard. However, I understand if you don't want to include this for readability.

Reply:
You could even further optimise this to avoid your jump statement:
```
size -= (size > 0);
return (size > 0) * entries[size].move;
```